pkg/svcat/service-catalog/broker.go

```diff
@@ -162,24 +172,44 @@ func (sdk *SDK) Register(brokerName string, url string, opts *RegisterOptions) (
 	return result, nil
 }
 
-// Sync or relist a broker to refresh its catalog metadata.
-func (sdk *SDK) Sync(name string, retries int) error {
-	for j := 0; j < retries; j++ {
-		catalog, err := sdk.RetrieveBroker(name)
-		if err != nil {
-			return err
+// Sync or relist a broker to refresh its broker metadata.
+func (sdk *SDK) Sync(name string, scopeOpts ScopeOptions, retries int) error {
+	success := false
+
+	for j := 0; j < retries && !success; j++ {
+		if scopeOpts.Scope.Matches(NamespaceScope) {
+			namespace := scopeOpts.Namespace
+			broker, err := sdk.RetrieveNamespacedBroker(namespace, name)
+			if err == nil {
+				broker.Spec.RelistRequests = broker.Spec.RelistRequests + 1
+
+				_, err = sdk.ServiceCatalog().ServiceBrokers(namespace).Update(broker)
+				if err != nil && !errors.IsConflict(err) {
+					return fmt.Errorf("could not sync service broker (%s)", err)
+				}
+				success = true
+				fmt.Printf("synced servicebroker (%s) in name space (%s)\n", name, scopeOpts.Namespace)
+			}
 		}
 
-		catalog.Spec.RelistRequests = catalog.Spec.RelistRequests + 1
-
-		_, err = sdk.ServiceCatalog().ClusterServiceBrokers().Update(catalog)
-		if err == nil {
-			return nil
-		}
-		if !errors.IsConflict(err) {
-			return fmt.Errorf("could not sync service broker (%s)", err)
+		if scopeOpts.Scope.Matches(ClusterScope) {
+			broker, err := sdk.RetrieveBroker(name)
+			if err == nil {
+				broker.Spec.RelistRequests = broker.Spec.RelistRequests + 1
+
+				_, err = sdk.ServiceCatalog().ClusterServiceBrokers().Update(broker)
+				if err != nil && !errors.IsConflict(err) {
```

Comment:
This consolidated error logic for both namespaced and cluster brokers doesn't properly handle when a non-conflict error occurs, which previously was returned. Essentially we only repeatedly try to sync on conflict errors, and we stop immediately on all other errors.

Reply:
ok, got it , will modify according to the suggestion here